packages/legend-application-studio/src/components/editor/edit-panel/mapping-editor/PropertyMappingsEditor.tsx

```diff
@@ -82,6 +94,14 @@ export const PropertyMappingsEditor = observer(
     isReadOnly: boolean;
   }) => {
     const { instanceSetImplementationState, property, isReadOnly } = props;
+    const [validationErrorMessage, setValidationErrorMessage] = useState(
```

Comment:
why do we need to make this a state, rather than just simply has

```ts
const validationErrorMessage = ...
```

Reply:
This validationErrorMessage is changing based on the property mappings and it also triggers if there is a red error.

Reply:
But why do we need to make this a component state and not just a normal variable?